internal/datacoord/compaction_task_clustering.go

```diff
@@ -558,7 +566,7 @@ func (t *clusteringCompactionTask) GetLabel() string {
 }
 
 func (t *clusteringCompactionTask) NeedReAssignNodeID() bool {
-	return t.GetState() == datapb.CompactionTaskState_pipelining && t.GetNodeID() == 0
+	return t.GetState() == datapb.CompactionTaskState_pipelining && (t.GetNodeID() == 0 || t.GetNodeID() == NullNodeID)
```

Comment:
dup with pr: https://github.com/milvus-io/milvus/pull/34537

Reply:
yes, I learn this from your pr